src/evaluate/evaluator/question_answering.py

```diff
@@ -0,0 +1,191 @@
+# Copyright 2022 The HuggingFace Datasets Authors and the TensorFlow Datasets Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Any, Callable, Dict, Optional, Tuple, Union
+
+# Lint as: python3
+from datasets import Dataset, load_dataset
+
+
+try:
+    from transformers import Pipeline, PreTrainedModel, PreTrainedTokenizer, TFPreTrainedModel
+
+    TRANSFORMERS_AVAILABLE = True
+except ImportError:
+    TRANSFORMERS_AVAILABLE = False
+
+from typing_extensions import Literal
+
+from ..module import EvaluationModule
+from ..utils.logging import get_logger
+from .base import Evaluator
+
+
+logger = get_logger(__name__)
+
+
+class QuestionAnsweringEvaluator(Evaluator):
+    """
+    Question answering evaluator.
+    This question answering evaluator can currently be loaded from [`evaluator`] using the default task name
+    `question-answering`.
+    Methods in this class assume a data format compatible with the [`QuestionAnsweringPipeline`].
+    """
+
+    def __init__(self, task="question-answering", default_metric_name=None):
```

Comment:
Yes I think it would be. For consistency, do you think we could put `accuracy` or `f1` in the text-classification and image-classification evaluators as default metric (they currently don't have one)?

Reply:
They (and this one) actually do. It is defined here:
https://github.com/huggingface/evaluate/blob/540956fb081cf909d29df02cb70dbab7db828b37/src/evaluate/evaluator/__init__.py#L33-L46

We should specify in the docstrings of the classes however, what the default is if `None` is provided.

Reply:
That's fine!